File: budget-planner-api/src/main/kotlin/beer/thierry/budgetplanner/api/model/user/UserDTO.kt
package beer.thierry.budgetplanner.api.model.user

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import java.util.*

data class UserDTO(
    var id: UUID = UUID.randomUUID(),
    @get:JvmName("getUsernameValue")
    var username: String = "",
    var email: String = "",
    var firstName: String = "",
    var lastName: String = "",
    var name: String = "",
    var profilePicture: String? = null,
    var authoritiesList: Collection<GrantedAuthority> = listOf(SimpleGrantedAuthority("ROLE_USER"))

) : UserDetails {
    override fun getAuthorities() = authoritiesList
    override fun getPassword() = null
    override fun getUsername() = username
    override fun isAccountNonExpired() = true
    override fun isAccountNonLocked() = true
    override fun isCredentialsNonExpired() = true
    override fun isEnabled() = true
}

File: centsible-ui/app/models/user/user-dto.ts
import type {Currency} from "~/models/budget-account/currency";

export interface UserDto {
  id: string;
  username: string;
  email: string;
  firstName: string;
  lastName: string;
  name: string;
  profilePicture?: string;
  locale: string;
  defaultCurrency: Currency;
  /** True when this user is the instance admin designated by the server configuration. */
  admin?: boolean;
}
